source/maths/mtx_transpose.c

```diff
@@ -0,0 +1,15 @@
+#include <c3d/maths.h>
+
+void Mtx_Transpose(C3D_Mtx* out)
+{
+	float swap;
+	for (int i = 0; i <= 2; i++)
```

Comment:
Are you sure this transposes a 4x4 matrix? This loop (and the one below it) suggests it's swapping a 3x3 matrix.